README.md

```diff
@@ -836,7 +836,7 @@ and if the data type is nullable, all when applicable.
     "example": "2017-06",
     "type": "string",
     "format": "year-month",
-    "pattern": "[0-9]{4}-[01][0-9]",
+    "pattern": "[0-9]{4}-(0?[1-9]|1[0-2])",
```

Comment:
This does not match `10` correctly.

Reply:
Actually, it will work fine if the entire input has to be consumed by the match

Reply:
But [0-9]{4}-(0?[1-9]|**1[0-2]**) does, right?

Reply:
A plain regex matches left-to-right eagerly, so `2024-10` would match `2024-1`, because `1` matches `0?[1-9]`. But if the regex requires the full input to be consumed. If the `2024-1` match was rejected because it does not include the `0`, then it would still end up matching `2024-10` with some extra backtracking in the end. Swapping the order to `(1[0-2]|0?[1-9])` would make it work in all cases, but it's not that relevant